lib/ecto/adapters/clickhouse/migration.ex

```diff
@@ -108,14 +108,37 @@ defmodule Ecto.Adapters.ClickHouse.Migration do
     [[drop | @conn.quote_table(index.prefix, index.name)]]
   end
 
-  def execute_ddl({command, %Constraint{} = _constraint})
+  def execute_ddl({command, %Constraint{} = constraint})
       when command in [:create, :create_if_not_exists] do
-    raise "TODO"
+    table_name = @conn.quote_table(constraint.prefix, constraint.table)
+
+    add =
+      case command do
+        :create -> " ADD CONSTRAINT "
+        :create_if_not_exists -> " ADD CONSTRAINT IF NOT EXISTS "
+      end
+
+    constraint_expr = constraint |> constraint_expr() |> Enum.join("")
```

Comment:
We probably don't need to `Enum.join` since we are returning an iolist anyway.

Reply:
Removed Enum.join as removed constraint_expr/1